meson.build

```diff
@@ -0,0 +1,1554 @@
+if javascript.found()
+    dependencies += javascript
+    sources += 'player/javascript.c'
+endif
+
+lcms2 = dependency('lcms2', version: '>= 2.6', required: get_option('lcms2'))
+if lcms2.found()
+    dependencies += lcms2
+endif
+
+libarchive = dependency('libarchive', version: '>= 3.4.0', required: get_option('libarchive'))
+if libarchive.found()
+    dependencies += libarchive
+    sources += ['demux/demux_libarchive.c',
+                'stream/stream_libarchive.c']
+endif
+
+libavdevice = dependency('libavdevice', version: '>= 57.0.0', required: get_option('libavdevice'))
+if libavdevice.found()
+    dependencies += libavdevice
+endif
+
+libbluray = dependency('libbluray', version: '>= 0.3.0', required: get_option('libbluray'))
+if libbluray.found()
+    dependencies += libbluray
+    sources += 'stream/stream_bluray.c'
+endif
+
+lua_version = [['52', '>= 5.2.0, < 5.3.0'],
+               ['lua52', '>= 5.2.0'],
+               ['lua5.2', '>= 5.2.0'],
+               ['lua-5.2', '>= 5.2.0'],
+               ['luajit', '>= 2.0.0'],
+               ['51', '>= 5.1.0, < 5.2.0'],
+               ['lua51', '>= 5.1.0'],
+               ['lua5.1', '>= 5.1.0'],
+               ['lua-5.1', '>= 5.1.0']]
+foreach version : lua_version
+    lua = dependency(version[0], version: version[1], required: get_option('lua'))
+    if lua.found()
+        break
+    endif
+endforeach
+if lua.found()
+    dependencies += lua
+    sources += 'player/lua.c'
+endif
+
+rubberband = dependency('rubberband', version: '>= 1.8.0', required: get_option('rubberband'))
+if rubberband.found()
+    dependencies += rubberband
+    sources += 'audio/filter/af_rubberband.c'
+endif
+
+sdl2 = dependency('sdl2', required: get_option('sdl2'))
+if sdl2.found()
+    dependencies += sdl2
+endif
+
+uchardet = dependency('uchardet', required: get_option('uchardet'))
+if uchardet.found()
+    dependencies += uchardet
+endif
+
+vapoursynth = dependency('vapoursynth', version: '>= 24', required: get_option('vapoursynth'))
+vapoursynth_script = dependency('vapoursynth-script', version: '>= 23', required: get_option('vapoursynth'))
+if vapoursynth.found() and vapoursynth_script.found()
+    dependencies += [vapoursynth, vapoursynth_script]
+    sources += 'video/filter/vf_vapoursynth.c'
+endif
+
+zimg = dependency('zimg', version: '>= 2.9', required: get_option('zimg'))
+if zimg.found()
+    dependencies += zimg
+    sources += ['video/filter/vf_fingerprint.c',
+                'video/zimg.c']
+endif
+
+
+# misc other dependencies
+dvbin = get_option('gpl').enabled() and not get_option('dvbin').disabled()
+if dvbin
+    sources += ['stream/dvb_tune.c',
+                'stream/stream_dvb.c']
+endif
+
+libm = cc.find_library('m', required: false)
+if libm.found()
+    dependencies += libm
+endif
+
+librt = cc.find_library('rt', required: false)
+if librt.found()
+    dependencies += librt
+endif
+
+if sdl2.found() and get_option('sdl2-gamepad')
```

Comment:
This will throw error `ERROR: Second argument to "and" is not a boolean.` when sdl2 is found. My suggestion is to enable all sdl2-gamepad/video/audio when sdl2 found

Reply:
Ah yeah, I need to change it to `not get_option('sdl2-gamepad`).disabled()`. I originally had some boolean options in the build file, but I though it was inconsistent to have some options take `true/false` while others were `auto/disabled/enabled` so I made everything a feature.